### PR TITLE
fix: Coalesce adjacent field labels

### LINKED ISSUE #9855: coalesce adjacent label fields
### Check for duplicates

- [ ] I have searched for similar issues before opening a new one.

### Problem

_No response_

### Request

some blocks visually have adjacent labels that are actually part of different inputs. e.g. first part of the string belongs to a dummy input and the second part belongs to a value input. this is due to various reasons with block design.

currently, all fields have their screenreader output separate by commas. but if two label fields are visually adjacent then there are extra commas.

we want to remove the extra commas between adjacent label fields (in other words, coalesce adjacent label fields into one "piece" of the screenreader output)

for the purpose of this issue a label field is any subclass of `FieldLabel` and "adjacent" means in subsequent field position on the block but respecting "rows" (e.g. a label field at the end of an end-row input is not adjacent to a label field in a statement input on the following row)

Example block definition:

```js
Blockly.Blocks['device_show_number'] = {
  init: function () {
    this.appendDummyInput()
      .appendField('show');

    this.appendValueInput('number')
      .setCheck('Number')
      .appendField('number');

    this.setPreviousStatement(true);
    this.setNextStatement(true);

    this.setColour('#1e90ff');
  }
};
```

### Alternatives considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by lizschwab**
## The basics
- [x] I [validated my changes](https://developers.google.com/blockly/guides/contribute/core#making_and_verifying_a_change)

### Resolves

Fixes https://github.com/RaspberryPiFoundation/blockly/issues/9855

### Proposed Changes

Coalesces adjacent field labels, taking into account whether the labels are on the same visual row. 

### Reason for Changes

Blocks built in ways where multiple labels are next to one another were computed as being comma separated when building out the block's ARIA label. This is confusing for users as these labels are treated as a single string for sighted users while being read as separate strings for screenreader users.

### Test Coverage

Unit tests have been added to cover the edge cases around this change.

### CODE REVIEW

File: packages/blockly/core/block_aria_composer.ts [hunk 1]
```diff
@@ -318,8 +318,83 @@ export function getInputLabels(
...
+      return label;
+    }
+    return inputLabel;
+  });
+}
+
+/**
+ * Returns whether an input's list of visible fields begins with a FieldLabel
+ *
+ * @internal
```
Comment (maribethb) [inline packages/blockly/core/block_aria_composer.ts]: You don't need to mark these as internal because they already aren't exported from this file

File: packages/blockly/tests/mocha/input_test.js [hunk 2]
```diff
@@ -379,19 +379,80 @@ suite('Inputs', function () {
...
+            .appendField('third')
+            .appendField('fourth');
+
+          const label = this.block.getAriaLabel();
+
+          assert.include(label, 'first second,');
+          assert.include(label, 'third fourth');
+        });
+        test('Statement inputs are not combined', function () {
+          const statementBlock = Blockly.Xml.domToBlock(
```
Comment (maribethb) [inline packages/blockly/tests/mocha/input_test.js]: prefer using json for these kinds of tests (even if older existing tests use xml).

**Review by maribethb — APPROVED** [review]
